### PR TITLE
Run shellcheck on bash completion

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: bash_completions.go [hunk 1]
```diff
@@ -226,12 +228,12 @@ __handle_word()
...
         return
     fi
-    __debug "${FUNCNAME}: c is $c words[c] is ${words[c]}"
+    __debug "${FUNCNAME[0]}: c is $c words[c] is ${words[c]}"
     if [[ "${words[c]}" == -* ]]; then
         __handle_flag
     elif __contains_word "${words[c]}" "${commands[@]}"; then
         __handle_command
-    elif [[ $c -eq 0 ]] && __contains_word "$(basename ${words[c]})" "${commands[@]}"; then
+    elif [[ $c -eq 0 ]] && __contains_word "$(basename "${words[c]}")" "${commands[@]}"; then
```
Comment (eparis) [inline bash_completions.go]: This seems odd and I don't know the right answer.  But now ${words[c]} is not in quotes?

File: bash_completions.go [hunk 2]
```diff
@@ -203,20 +205,20 @@ __handle_noun()
...
-    __debug "${FUNCNAME}: c is $c words[c] is ${words[c]}"
+    __debug "${FUNCNAME[0]}: c is $c words[c] is ${words[c]}"
 
     local next_command
     if [[ -n ${last_command} ]]; then
         next_command="_${last_command}_${words[c]//:/__}"
     else
         if [[ $c -eq 0 ]]; then
-            next_command="_$(basename ${words[c]//:/__})"
+            next_command="_$(basename "${words[c]//:/__}")"
```
Comment (eparis) [inline bash_completions.go]: This seems to unquote the variable....

File: bash_completions.go [hunk 3]
```diff
@@ -88,11 +88,13 @@ __handle_reply()
...
                 if [[ ${index} -ge 0 ]]; then
-                    COMPREPLY=() PREFIX="" cur="${cur#*=}"
+                    COMPREPLY=()
+                    PREFIX=""
+                    cur="${cur#*=}"
                     ${flags_completion[${index}]}
                     if [ -n "${ZSH_VERSION}" ]; then
                         # zfs completion needs --flag= prefix
-                        eval COMPREPLY=( "\${COMPREPLY[@]/#/${flag}=}" )
+                        eval "COMPREPLY=( \"\${COMPREPLY[@]/#/${flag}=}\" )"
```
Comment (eparis) [inline bash_completions.go]: And this just melts my mind....

File: bash_completions.go (hunk 2 above)
Comment (sttts) [inline bash_completions.go]: What do you mean? The `$(...)` creates a new "quote context".

File: bash_completions.go (hunk 3 above)
Comment (sttts) [inline bash_completions.go]: single quotes might be more suitable.

File: bash_completions.go (hunk 1 above)
Comment (sttts) [inline bash_completions.go]: As below, `"$(..."$X"...)"` is correct. It does not "unquote" the variable.

File: bash_completions.go (hunk 3 above)
Comment (sttts) [inline bash_completions.go]: We need the `${flag}` interpolation. Single quotes don't make it easier.

**Pull request merged**